### ISSUE TITLE
Distribution for p(z|y)?

### ISSUE THREAD

**Issue by abdulfatir** [CompVis/vunet#15]
Hi

In the paper, you define q(z|x,y) as Gaussian but I don't see a similar definition for p(z|y). What is p(z|y)? Is it Gaussian too?

**Comment by pesser** [comment]
Hi,

yes the prior is modeled as a Gaussian as well. Thanks for pointing it out, we should probably add it in an updated version of the paper!

**Comment by abdulfatir** [comment]
Thanks for a swift response. I have a follow-up. I haven't studied the code in detail but it looks like [`latent_kl`](https://github.com/CompVis/vunet/blob/master/models.py#L217) computes the **KL(q(z|x,y)||p(z|y))**; can you explain how it is working?

```python
def latent_kl(q, p):
    mean1 = q
    mean2 = p

    kl = 0.5 * tf.square(mean2 - mean1)
    kl = tf.reduce_sum(kl, axis = [1,2,3])
    kl = tf.reduce_mean(kl)
    return kl
```

I see that you're taking the squared difference.

**Comment by pesser** [comment]
The KL divergence between two gaussians with fixed covariance is equal to the squared difference of their means (up to constant factors). Sometimes variances are estimated too but I found no advantage in doing it here, in fact it decreased stability of training. See also #4 and #5.

**Comment by abdulfatir** [comment]
Got it. Didn't realize it was fixed variance. Thanks. :)
